### PR TITLE
Expands "genderless" to all species, adds body-type option, removes some gender checks

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Continues https://github.com/ParadiseSS13/Paradise/pull/23469, and fundamentally revives the changes made in https://github.com/ParadiseSS13/Paradise/pull/16007.

Expands the "genderless" option to work across all species, beyond the few that already supported it. If you select this option, you will be referred to by most checks as they/them.
On that note, this PR also adds a body-type option, between masculine and feminine, which allows you to change your base body sprite.
This means that you can have a feminine body with male pronouns, or a masculine bodytype with plural pronouns.

This PR also does away with some checks based on gender, or associates them with bodytype instead of gender. I've made facial hair and all hairstyles accessible to both bodytypes, although undergarments are still limited per bodytype owing to sprite differences.

## Why It's Good For The Game
This provides players with a lot more flexibility and control over how they might want to design their characters with appearance, how they present themselves, and how the game refers to them. This is a fantasy space simulator in the 2500s, at which point people probably do have a lot of control over their body.

Genderless species are also nothing new, with vox already having the option to be genderless, male, or female. On top of that, skrell have already been seemingly able to use both male and female hairstyles, which really begs into question why we don't have this for everyone.

Lastly, Paradise is kind of lagging behind its peers in supporting more advanced and inclusive customization options such as this. There are a fair amount of people in the SS13 community who have complicated relationships with gender, and it's a positive change to allow people to create the characters that they want to create. Even /tg/station, with its past reputation, has beaten us to the punch in this regard. It's about time for us to catch up.

## Images of changes
image
image

## Testing
Created new characters with different attributes, verified that they loaded and saved correctly, and presented correctly in-game.

## Changelog
:cl: SabreML, Daylight2
tweak: All species now have access to the "Genderless" option
tweak: You can now choose your character's body-type, between masculine and feminine
tweak: Hairstyles and facial hair are now all accessible regardless of gender or body-type
/:cl:

### CODE REVIEW

**Comment by Alecksohs** [comment]
based

**Comment by SynthTwo** [comment]
Finally. It's 2024, it's about time. If anyone concerned about players trolling with this, do we even want those type of players in the server to begin with?

File: SQL/updates/53-54.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Updating DB from 23-24,
```
Comment (AffectedArc07) [inline SQL/updates/53-54.sql:1]: You either mistyped on the numpad or grabbed a REALLY old file

File: config/example/config.toml [hunk 2]
```diff
@@ -66,40 +64,52 @@ enable_localhost_autoadmin = true
...
+    "+POSSESS",
+    "+PROCCALL",
+    "+VAREDIT",
+    "+SOUND",
+    "+SPAWN",
+  ]},
+  {name = "Head of Staff", rights = ["+EVERYTHING"]},
+  {name = "Hosting Provider", rights = ["+EVERYTHING"]},
+  {name = "Maintainers", rights = ["+EVERYTHING"]},
 ]
```
Comment (AffectedArc07) [inline config/example/config.toml:88]: Unformat this please

File: SQL/updates/53-54.sql (hunk 1 above)
Comment (lewcc) [reply SQL/updates/53-54.sql:1]: Yeah, that's a result of me using sabre's old file, oops! I'll fix that

File: code/__HELPERS/sanitize_values.dm [hunk 3]
```diff
@@ -23,15 +23,15 @@
 
 
 //more specialised stuff
-/proc/sanitize_gender(gender,neuter=0,plural=0, default="male")
+/proc/sanitize_gender(gender, neuter=FALSE, default="male")
```
Comment (DGamerL) [inline code/__HELPERS/sanitize_values.dm:26]:
```suggestion
/proc/sanitize_gender(gender, neuter = FALSE, default = "male")
```

File: code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm [hunk 4]
```diff
@@ -33,7 +33,7 @@
 				L[D.name] = D
 				full_list[D.name] = D
 
-			switch(D.gender)
+			switch(D.body_type)
 				if(MALE)	male[D.name] = D
 				if(FEMALE)	female[D.name] = D
```
Comment (DGamerL) [inline code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm:38]: Please quickly make this code compliant with our COC

**Comment by TORE01** [comment]
Having a "none of the above" option will always be useful because you can pick it and just be whatever you want to be. But giving human or any other biological species the option to be something in between XX and XY is just silly if you consider the setting of the game we're playing. Science has gone extremely far to the point where you can change between XX and XY with the help of a button. Why would this even be a thing? It's just going to add confusion and unneseccary drama.

> Finally. It's 2024, it's about time. If anyone concerned about players trolling with this, do we even want those type of players in the server to begin with?

It's 2565 my friend. Embrace it.

File: code/modules/client/preference/character.dm [hunk 5]
```diff
@@ -1803,6 +1810,7 @@
 	character.gen_record = gen_record
 
 	character.change_gender(gender)
+	character.body_type = body_type  // TODO does this update the character properly or do we need a setter here
```
Comment (Contrabang) [inline code/modules/client/preference/character.dm:1813]: todo

File: code/modules/mob/living/carbon/carbon_defines.dm [hunk 6]
```diff
@@ -1,5 +1,7 @@
 /mob/living/carbon
 	gender = MALE
+	/// The body type associated with the body
+	var/body_type = MALE
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/carbon_defines.dm:4]: I don't really see this being used for anything but humans, is there a good reason to have it here?

File: code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm [hunk 7]
```diff
@@ -581,7 +583,7 @@
 /datum/sprite_accessory/socks/thin_knee
 	name = "Knee-high Thin"
 	icon_state = "thin_knee"
-	gender = FEMALE
+	body_type = FEMALE
```
Comment (Contrabang) [inline code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm:586]: I wonder if these are still necessary, servers like TG have them applicable to all genders, and I don't think they have gender specific sprites

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm (hunk 7 above)
Comment (lewcc) [reply code/modules/mob/new_player/sprite_accessories/sprite_accessories.dm:586]: It's necessary for at least some undergarments, where the differences in body shape mean the sprites don't fit *quite* right. Socks and stuff are probably fine for both.

File: code/modules/client/preference/character.dm (hunk 5 above)
Comment (lewcc) [reply code/modules/client/preference/character.dm:1813]: It did seem to, yeah.

**Review by Contrabang — APPROVED** [review]
looks good to me, still needs AA approval

**Comment by AffectedArc07** [comment]
I'll handle this over the weekend.

**Comment by lewcc** [comment]
!merge_upstream

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]
Where are the changes to `20-load_characters.dm` to actually load the characters from the DB?

**Review by AffectedArc07 — APPROVED** [review]
Welp, here we go

**Review by Contrabang — APPROVED** [review]
# SHIP IT!!!!

**Comment by SabreML** [comment]
woo!

**Pull request merged**